WordPress/gutenberg
# Testing: Remove unnecessary Enzyme React 16 workarounds

Closes #11214 

This pull request seeks to update unit tests which had implemented workarounds to a previous lack of support for React 16 features, namely context and `forwardRef`.

Specifically:

- Drops a mock implementation of the `Button` component used in unwrapping `forwardRef`
- Un-skips a `BlockControls` snapshot test

In so doing, it bumps `enzyme-adapter-react-16` (once more from #13922) to the latest 1.10.0 release.

**Testing instructions:**

Ensure unit tests pass:

```
npm run test-unit
```

Ensure snapshot changes are sensible.
